source/material_model/drucker_prager.cc

```diff
@@ -0,0 +1,388 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/drucker_prager.h>
+#include <deal.II/base/parameter_handler.h>
+
+using namespace dealii;
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+
+    template <int dim>
+    double
+    DruckerPrager<dim>::
+    viscosity (const double temperature,
+               const double pressure,
+               const std::vector<double> &composition,
+               const SymmetricTensor<2,dim> &strain_rate,
+               const Point<dim> &position) const
+    {
+      // For the very first time this function is called,
+      // we prescribe a representative reference strain rate.
```

Comment:
what is the reason to doing this? Is it because the strain rate is zero and the following computations make no sense otherwise? I think it would help to add a comment *why* things are done, not just *how*.